crates/oxc_semantic/src/symbol.rs

```diff
@@ -62,10 +62,13 @@ impl SymbolTable {
         self.spans.iter_enumerated().map(|(symbol_id, _)| symbol_id)
     }
 
+    // Symbol are created in dfs order, so we make tow assumptions:
+    // 1. The start of [SymbolTable.spans] is strictly monotonically increasing.
+    // 2. There are no two symbols with the same span.
+    // So we could use binary search to find the symbol id.
     pub fn get_symbol_id_from_span(&self, span: Span) -> Option<SymbolId> {
-        self.spans
-            .iter_enumerated()
-            .find_map(|(symbol, &inner_span)| if inner_span == span { Some(symbol) } else { None })
+        let index = self.spans.raw.binary_search_by(|a| a.start.cmp(&span.start)).ok()?;
```

Comment:
i wonder whether a debug assert would be a good idea here to validate the assumptions